[Fix] 'std::out_of_range' error

AI-Pranto opened the pull request:
Clean PR :)

- Simple enough to fix `terminate called after throwing an instance of ‘std::out_of_range’` error.
- **@gridley's comment**: Before merging though, can we rename this to "max_surface_particles"? The name used here is really ambiguous.

gridley commented:
Oh, OK, you reopened a separate PR, nice. Reviewing now...

gridley reviewed (approved):
Looks good to me! I think we should get someone else's opinion as well before merging though.

paulromano commented:
On the naming, I think the "surface" in `max_surface_particles` is actually a little redundant. This is already part of the `surf_source_write` attribute, so to me it's clear that `max_particles` means "maximum number of particles to use for surface source write". @ypark234 Since this was your feature, do you have an opinion on the naming?

gridley commented:
> On the naming, I think the "surface" in `max_surface_particles` is actually a little redundant. This is already part of the `surf_source_write` attribute, so to me it's clear that `max_particles` means "maximum number of particles to use for surface source write". @ypark234 Since this was your feature, do you have an opinion on the naming?

I agree with this... In the settings.cpp part is where I mainly thought the description should be included. So the changes to the code should actually be pretty small with no changes to any XML formatting.

paulromano commented:
@gridley Ah ok! Maybe that's what you were suggesting originally? I agree with you though -- changing the variable name on the C++ side makes sense, but we shouldn't change the user input.

AI-Pranto commented:
> but we shouldn't change the user input.

@gridley Do you agree with @paulromano?

gridley commented:
Yeah @AI-Pranto, since in the input file, it says `max_particles` under the surface tally XML element, but in the C++ side, it's just a variable called `max_particles` that is not in a place giving it context that it relates to surface sources. Although a comment is there, when it's used in other places it's not really clear that it relates to surfaces IMO.

ypark234 commented:
@paulromano @gridley - I agree both: On user input side, 'surface' might be redundant. On C++ side, having 'surface' in the variable name is good so that it relates to surface source feature. Thanks for pointing it out!

paulromano reviewed (changes requested):
@AI-Pranto It looks like the user input is still modified here. settings.py. settings.rst, settings.rng, and test_settings.py should all be unmodified.

AI-Pranto commented:
>  It looks like the user input is still modified here. settings.py. settings.rst, settings.rng, and test_settings.py should all be unmodified.

@paulromano I have addressed your feedback.
